scripts/globals/geo.lua

```diff
@@ -0,0 +1,26 @@
+-----------------------------------
+-- Geomancer helpers
+-----------------------------------
+require("scripts/globals/pets")
+require("scripts/globals/status")
+-----------------------------------
+
+tpz = tpz or {}
+tpz.geo = tpz.geo or {}
+
+tpz.geo.spawnLuopan = function(player, target, aura_effect, tick_effect, tick_power, target_type, spell_cost)
+
+    tpz.pet.spawnPet(player, tpz.pet.id.LUOPAN)
+    local luopan = player:getPet()
+    local modelID = aura_effect
```

Comment:
I'm not going to fight the passed in model argument _for now_, but models _shouldn't_ be an argument for `spawnLuopan` in the Geo- spell scripts. Luopan models are just base model ID (depending if they're supportive or offensive) + the spell's element. When our spell elements are fixed at the fundamental level, all models can be generated by `spawnLuopan` itself. (Although _how_ we inform spawnLuopan of the luopan's intended element will be a _different_ matter.

Reply:
I've added some big comments explaining how this should work in the future 👍